src/fparser/two/tests/fortran2008/test_error_stop_stmt_r856.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2022 Science and Technology Facilities Council
+
+# All rights reserved.
+
+# Modifications made as part of the fparser project are distributed
+# under the following license:
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Test Fortran 2008 rule R856
+
+    error-stop-stmt is ERROR STOP [ stop-code ]
+'''
+
+import pytest
+from fparser.two.utils import NoMatchError
+from fparser.two.Fortran2008 import Error_Stop_Stmt
+
+
+def test_simple(f2008_create):
+    result = Error_Stop_Stmt('ERROR STOP')
+    assert str(result) == 'ERROR STOP'
+
+
+def test_stop_code(f2008_create):
+    result = Error_Stop_Stmt('ERROR STOP 1')
+    assert str(result) == 'ERROR STOP 1'
+
+
+def test_error1(f2008_create):
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = Error_Stop_Stmt('ERROR () STOP')
+    assert "Error_Stop_Stmt: 'ERROR () STOP'" in str(excinfo.value)
```

Comment:
Please could you also add more of a functional test to demonstrate that the statement is correctly matched when is appears in e.g. a subroutine.

Reply:
Thanks for suggesting this, turns out flushing this change through to Action_Stmt requires re-implementing a significant number of Fortran2003 classes because of associated constraint classes and the renumbering of those... Hope to have that ready by this afternoon.

Reply:
Ooph, I have to admit I wasn't expecting that! Thanks for carrying it through.